src/sinks/util/unix.rs

```diff
@@ -176,7 +178,15 @@ where
             let _open_token = OpenGauge::new().open(|count| emit!(ConnectionOpen { count }));
 
             let result = match sink
-                .send_all_peekable(&mut (&mut input).map(|item| item.item).peekable())
+                .send_all_peekable(
+                    &mut (&mut input)
+                        .map(|item| {
+                            drop(item.finalizers);
+                            self.acker.ack(1);
+                            item.item
```

Comment:
I think this is acking the item before it's sent and not after. Having said that, I'm not sure how to easily make this ack after.

Reply:
@jszwedko and I discussed this and agreed to push the remaining work for it to a separate issue to allow this one to close out. I'll write up that issue shortly.